### PR TITLE
SGX target: implemented vectored I/O

### PR DESCRIPTION
**Pull request by jethrogb**
r? @sfackler 

Includes #59857

### CODE REVIEW

**Comment by jonas-schievink** [comment]
#58957 does not look like the right link

**Comment by Dylan-DPC** [comment]
ping from triage @sfackler waiting for your review on this

**Review by sfackler — COMMENTED** [review]
Oops, sorry!

File: src/libstd/sys/sgx/abi/usercalls/mod.rs [hunk 1]
```diff
@@ -18,6 +19,29 @@ pub fn read(fd: Fd, buf: &mut [u8]) -> IoResult<usize> {
     }
 }
 
+/// Scatter variant of usercall `read`. This will do a single `read` usercall
+/// and scatter the read data among `bufs`.
+#[unstable(feature = "sgx_platform", issue = "56975")]
+pub fn read_vectored(fd: Fd, bufs: &mut [IoVecMut<'_>]) -> IoResult<usize> {
```
[the remark this answers is not in the document]
Comment (jethrogb) [reply src/libstd/sys/sgx/abi/usercalls/mod.rs]: Done

**Comment by sfackler** [comment]
r=me after a rebase

**Comment by jethrogb** [comment]
Rebased

**Comment by jethrogb** [comment]
Um

**Comment by jethrogb** [comment]
Ok I think bors was just slow to comment that it's trying this PR and nothing bad actually happened!

**Pull request merged**